### DISCUSSION TITLE
Using create_model with a list of child models.

### DISCUSSION THREAD

**Discussion by chrishemmings** [pydantic/pydantic#4548] [Question] [answered]
Hey there,

Firstly, Pydantic is awesome, so many thanks for everybody's hard work on it 👍 

I'm trying the use the `create_model()` method to turn a `dict` into a Pydantic model.  All is good generally, but, when I want to define a list of another model I get an error.  For example, using this;

```python
child = {
    'c': (str, ...),
    'd': (str, ...),
}

child_model = create_model('child', **child)

parent = {
    'a': (str, ...),
    'b': (List[child_model], ...)
}

parent_model = create_model('child', **parent)
```

I see the error `KeyError: <class 'pydantic.main.child'>`   I'm using Pydantic 1.10.2, any pointers on how to get over this issue would be much appreciated.

**Comment by samuelcolvin** [comment, marked as answer]
> Firstly, Pydantic is awesome, so many thanks for everybody's hard work on it +1

thanks so much for the kind words, and for trying pydantic.

Your exact code works fine for me, complete example:

```py
from typing import List

from pydantic import create_model, __version__

print('pydantic version:', __version__)

child = {
    'c': (str, ...),
    'd': (str, ...),
}

child_model = create_model('child', **child)

parent = {
    'a': (str, ...),
    'b': (List[child_model], ...)
}

parent_model = create_model('parent', **parent)

m = parent_model(a='a', b=[{'c': 'c', 'd': 'd'}])
print(repr(m))
#> parent(a='a', b=[child(c='c', d='d')])
```

My guess is your problem is either to do with `from __future__ import annotations` and scope, or importing `child`. But hard to know without a full example.

**Reply by chrishemmings** [reply]
Thanks for you reply @samuelcolvin .  Not sure what was going on, so, completely refactored that bit of functionality and all is well.  Very odd!

Thanks again for the confirmation 👍
